src/test/components/MarkerChart.test.js

```diff
@@ -97,87 +99,154 @@ function setupWithProfile(profile) {
+    const {
+      flushRafCalls,
+      dispatch,
+      markerChart,
+      flushDrawLog,
+    } = setupWithProfile(profile);
 
     dispatch(changeSelectedTab('marker-chart'));
     markerChart.update();
-    expect(markerChart).toMatchSnapshot();
+    flushRafCalls();
+    flushDrawLog();
+
+    // No tooltip displayed yet
+    expect(markerChart.find('Tooltip').exists()).toEqual(false);
+
+    // Move the mouse on top of an item.
+    markerChart.find('canvas').simulate('mousemove', {
+      nativeEvent: { offsetX: 50, offsetY: 5 },
+      pageX: 50,
+      pageY: 5,
+    });
+    markerChart.update();
+    flushRafCalls();
+
+    const drawCalls = flushDrawLog();
+    expect(drawCalls).toMatchSnapshot();
+
+    // The tooltip should be displayed
+    expect(markerChart.find('Tooltip').exists()).toEqual(true);
   });
 
-  it('shows a reason when a profil has no network markers', () => {
-    const profile = getProfileWithMarkers(MARKERS);
-    const { dispatch, markerChart } = setupWithProfile(profile);
+  describe('with search strings', function() {
+    function getFillTextCalls(drawCalls) {
+      return drawCalls
+        .filter(([methodName]) => methodName === 'fillText')
+        .map(([_, text]) => text);
+    }
+
+    const searchString = 'Dot marker E';
+
+    it('renders lots of markers initially', function() {
+      const profile = getProfileWithMarkers(MARKERS);
+      const { flushRafCalls, flushDrawLog } = setupWithProfile(profile);
+
+      flushRafCalls();
+      const text = getFillTextCalls(flushDrawLog());
+      expect(text.length).toBeGreaterThan(1);
+      // Check that our test search string is in here:
+      expect(text.filter(t => t === searchString).length).toBe(1);
+    });
+
+    it('renders only the marker that was searched for', function() {
+      const profile = getProfileWithMarkers(MARKERS);
+      const {
+        flushRafCalls,
+        dispatch,
+        flushDrawLog,
+        markerChart,
+      } = setupWithProfile(profile);
+
+      // Flush out any existing draw calls.
+      flushRafCalls();
+      flushDrawLog();
+
+      // Update the chart with a search string.
+      dispatch(changeMarkersSearchString(searchString));
+      markerChart.update();
+      flushRafCalls();
+
+      const text = getFillTextCalls(flushDrawLog());
+      expect(text.length).toBe(1);
+      expect(text[0]).toBe(searchString);
+    });
+  });
 
-    dispatch(changeSelectedTab('network-chart'));
-    markerChart.update();
-    expect(markerChart).toMatchSnapshot();
+  describe('EmptyReasons', () => {
+    it('shows a reason when a profile has no markers', () => {
+      const profile = getProfileWithMarkers([]);
+      const { dispatch, markerChart } = setupWithProfile(profile);
+
+      dispatch(changeSelectedTab('marker-chart'));
+      markerChart.update();
+      expect(markerChart.find(EmptyReasons)).toMatchSnapshot();
+    });
+
+    fit("shows a reason when a profile's markers have been filtered out", () => {
```

Comment:
ooops, you left a "fit" here, as a result all other tests are skipped :) You could see this because a lot of snapshots were removed.